c/json.c

```diff
@@ -1020,36 +1042,46 @@ JsonToken *getStringToken(JsonTokenizer *tokenizer) {
     } else if (lookahead == '\\') {
       int backSlash = jsonTokenizerRead(tokenizer);
       int c = jsonTokenizerRead(tokenizer);
-      if (pos < tokenizer->bufferSize) {
-        switch (c) {
-          case 'b':
-            buffer[pos++] = '\b';
-            break;
-          case 'n':
-            buffer[pos++] = '\n';
-            break;
-          case 'f':
-            buffer[pos++] = '\f';
-            break;
-          case 'r':
-            buffer[pos++] = '\r';
-            break;
-          case 't':
-            buffer[pos++] = '\t';
-            break;
-          default:
-            buffer[pos++] = c;
+      if (pos >= tokenizer->bufferSize) {
+        int growRc = jsonTokenizerGrowBuffer(tokenizer);
+        if (growRc < 0) {
+          break;
         }
       }
+      switch (c) {
+        case 'b':
+          tokenizer->buffer[pos++] = '\b';
+          break;
+        case 'n':
+          tokenizer->buffer[pos++] = '\n';
+          break;
+        case 'f':
+          tokenizer->buffer[pos++] = '\f';
+          break;
+        case 'r':
+          tokenizer->buffer[pos++] = '\r';
+          break;
+        case 't':
+          tokenizer->buffer[pos++] = '\t';
+          break;
+        default:
+          tokenizer->buffer[pos++] = c;
+      }
     } else {
       int c = jsonTokenizerRead(tokenizer);
-      if (isprint(c) && pos < tokenizer->bufferSize) {
-        buffer[pos++] = c;
+      if (isprint(c)) {
+        if (pos >= tokenizer->bufferSize) {
+          int growRc = jsonTokenizerGrowBuffer(tokenizer);
+          if (growRc < 0) {
+            break;
```

Comment:
I would ask you to introduce a new token type denoting a too long string. The token would help JSON parser to issue a correct error message.

Reply:
done